frontend/react/style-guide.md

```diff
@@ -1351,6 +1353,121 @@ const { isLoading, isError, error, isSuccess, data, status, fetchStatus } = useQ
   >- The **`fetchStatus`** gives information about the **`queryFn`**: Is it running or not?
 
 
+## Context
+
+### Introduction
+Context provides a way to share values like these between components without having to explicitly pass a prop through every level of the tree.
+>For advanced information on Context, check [here](https://tkdodo.eu/blog/practical-react-query).
+>Official Documentation, check [here](https://reactjs.org/docs/context.html)
+
+### `contextFactory`
+First of all we have a HOC called ProviderWrapper that we are going to use in the base component of our context. This component receives a context, a reducer and an initial state which we should create inside another file, where we will have everything related to this, with the help of the `contextFactory` function located in `config/context.ts`.
+
+```tsx
+import { createContext, useContext, Dispatch } from 'react';
+
+export const contextFactory = <State, Action>(initialState: State) => {
+  interface Store {
+    state: State;
+    dispatch: Dispatch<Action>;
+  }
+
+  const Context = createContext<Store>({
+    state: { ...initialState },
+    // eslint-disable-next-line @typescript-eslint/no-empty-function, no-empty-function
+    dispatch: () => {}
+  });
+
+  const useSelector = <T>(selector: (arg: State) => T) => {
+    const { state } = useContext(Context);
+    return selector(state);
+  };
+
+  const useDispatch = () => {
+    const { dispatch } = useContext(Context);
+    return dispatch;
+  };
+
+  // eslint-disable-next-line @typescript-eslint/naming-convention
```

Comment:
I think this comments could be self-defeating

Reply:
It has been removed :godmode: